core/src/main/java/com/gentics/mesh/plugin/manager/MeshPluginManagerImpl.java

```diff
@@ -588,4 +574,55 @@ public PluginResponse toResponse(MeshPlugin plugin) {
 		return response;
 	}
 
+	protected void startPluginsFrom(List<PluginWrapper> source) {
+		List<PluginWrapper> sourceCopy;
+		
+		if (source == failedPlugins) {
+			sourceCopy = new ArrayList<>(source);
+		} else {
+			sourceCopy = source;
+		}
+		
+		for (PluginWrapper pluginWrapper : sourceCopy) {
+			PluginState pluginState = pluginWrapper.getPluginState();
+			if ((PluginState.DISABLED != pluginState) && (PluginState.STARTED != pluginState)) {
+				try {
+					log.info("Start plugin '{}'", getPluginLabel(pluginWrapper.getDescriptor()));
+
+					Plugin plugin = pluginWrapper.getPlugin();
+
+					//if (plugin.isUsingRestClient()) {
```

Comment:
I would really like to limit the application of the topolock check to the plugins, that do use REST API (i.e. DB access), although I see no way to check this before the plugin is initialized. Say, if the plugin requests MeshRestClient module, then we know it to potentially change the DB, but this is done on the plugin init the earliest, but that's too late for us. Question is open though.